_glaredb/sq-functions/parquet_scan.md

```diff
@@ -6,37 +6,163 @@ parent: SQL functions
 
 # `parquet_scan`
 
-Read a parquet file. The file may exist locally or remotely.
+Read parquet files from a local filesystem or supported object stores.
+
+URLs provided to `parquet_scan` may also include globbing characters to allow
+scanning multiple files when reading from S3, GCS, or a local filesystem.
+Globbing is not allowed when querying parquet files over HTTP/HTTPS.
 
 ## Syntax
 
 ```sql
+-- Single url or path.
 parquet_scan(<url>)
+-- Multiple urls or paths.
+parquet_scan([<url>])
+-- Using a cloud credentials object.
+parquet_scan(<url>, <credential_object>)
+-- Required named argument for S3 buckets.
+parquet_scan(<url>, <credentials_object>, region => '<aws_region>')
+-- Pass S3 credentials using named arguments.
+parquet_scan(<url>, access_key_id => '<aws_access_key_id>', secret_access_key => '<aws_secret_access_key>', region => '<aws_region>')
+-- Pass GCS credentials using named arguments.
+parquet_scan(<url>, service_account_key => '<gcp_service_account_key>')
+```
+
+Parameter descriptions and which object store it's relevant to.
+
+| Parameter                 | Object store | Description                                                                |
+|---------------------------|--------------|----------------------------------------------------------------------------|
+| `url`                     | All          | The URL or path to a parqet file to scan.                                  |
+| `credential_object`       | S3 and GCS   | A database object storing credentials for accessing the object or objects. |
+| `aws_region`              | S3           | If scanning an object in S3, the region of the bucket.                     |
+| `aws_access_key_id`       | S3           | ID of AWS access key with permissions to read from the bucket.             |
+| `aws_secret_access_key`   | S3           | Secret associated with the AWS access key.                                 |
+| `gcp_service_account_key` | GCS          | A JSON-encoded GCP service account key with access to the bucket.          |
+
+## Usage
+
+### Local files
+
+Local files can be read by passing the path or a globbed path to `parquet_scan`.
+Paths may be absolute or relative.
+
+```sql
+-- Read a relative path.
+SELECT * FROM parquet_scan('./my_data.parquet');
+-- Read all parquet files in a directory.
+SELECT * FROM parquet_scan('./directory_of_data/*.parquet');
+-- Read multiple explicitly provided files.
+SELECT * FROM parquet_scan(['./directory_of_data/1.parquet', './directory_of_data/2.parquet']);
+```
```

Comment:
it might be nice to add multiple dirs example, since it requires `**` and that's not obvious necessarily